compiler/rustc_hir_analysis/src/hir_ty_lowering/lint.rs

```diff
@@ -28,6 +24,17 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             return;
         };
 
+        let in_path = match tcx.parent_hir_node(self_ty.hir_id) {
```

Comment:
I'm pretty sure this needs to handle paths in `Node::Pat`, too.

IINM, not doing so regresses the suggestion for the following code:

```rs
trait Trait {}
impl dyn Trait { const CONST: () = (); }

fn main() {
    match () { Trait::CONST => {} }
}
```

Should keep suggesting `<dyn Trait>::CONST` inside the match.

Reply:
Thanks for catching that! I love how we didn't have a test for that 💀